test/test_pdfs/test_gaussian_pdf_gradient.C

```diff
@@ -0,0 +1,112 @@
+//-----------------------------------------------------------------------bl-
+//--------------------------------------------------------------------------
+//
+// QUESO - a library to support the Quantification of Uncertainty
+// for Estimation, Simulation and Optimization
+//
+// Copyright (C) 2008-2015 The PECOS Development Team
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the Version 2.1 GNU Lesser General
+// Public License as published by the Free Software Foundation.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc. 51 Franklin Street, Fifth Floor,
+// Boston, MA  02110-1301  USA
+//
+//-----------------------------------------------------------------------el-
+
+#include <queso/Environment.h>
+#include <queso/GslVector.h>
+#include <queso/GslMatrix.h>
+#include <queso/GaussianJointPdf.h>
+
+#define TOL 1e-15
+
+int main(int argc, char ** argv)
+{
+#ifdef QUESO_HAS_MPI
+  MPI_Init(&argc, &argv);
+
+  QUESO::FullEnvironment env(MPI_COMM_WORLD, "", "", NULL);
+#else
+  QUESO::FullEnvironment env("", "", NULL);
+#endif
+
+  unsigned int dim = 3;
+  QUESO::VectorSpace<> paramSpace(env, "param_", dim, NULL);
+
+  QUESO::GslVector paramMins(paramSpace.zeroVector());
+  paramMins.cwSet(-INFINITY);
+
+  QUESO::GslVector paramMaxs(paramSpace.zeroVector());
+  paramMaxs.cwSet(INFINITY);
+
+  QUESO::BoxSubset<> paramDomain("param_", paramSpace, paramMins, paramMaxs);
+
+  QUESO::GslVector mean(paramSpace.zeroVector());
+  QUESO::GslMatrix var(paramSpace.zeroVector());
+  mean[0] = 2.0;
+  mean[1] = 3.0;
+  mean[2] = 4.0;
+  var(0,0) = 5.0;
+  var(1,1) = 6.0;
+  var(2,2) = 7.0;
+
+  // Check the zero-gradient case
+  QUESO::GaussianJointPdf<> pdf("", paramDomain, mean, var);
+
+  QUESO::GslVector lnGradVector(paramSpace.zeroVector());
+  QUESO::GslVector gradVector(paramSpace.zeroVector());
+  QUESO::GslVector point(mean);
+
+  double lnPdfValue1 = pdf.lnValue(point, NULL, &lnGradVector, NULL, NULL);
```

Comment:
It's tested [here](https://github.com/libqueso/queso/blob/dev/test/test_GaussianVectorRVClass/test_VectorPdf_gsl.C#L54).